updater/lib/dependabot/job.rb

```diff
@@ -43,87 +45,132 @@ class Job
       dependency_groups
       dependency_group_to_refresh
       repo_private
-    ).freeze
-
-    attr_reader :allowed_updates,
-                :credentials,
-                :dependencies,
-                :existing_pull_requests,
-                :existing_group_pull_requests,
-                :id,
-                :ignore_conditions,
-                :package_manager,
-                :requirements_update_strategy,
-                :security_advisories,
-                :security_updates_only,
-                :source,
-                :token,
-                :vendor_dependencies,
-                :dependency_groups,
-                :dependency_group_to_refresh
+    ).freeze, T::Array[Symbol])
+
+    sig { returns(T::Array[T::Hash[String, T.untyped]]) }
+    attr_reader :allowed_updates
+
+    sig { returns(T::Array[T.untyped]) }
```

Comment:
```suggestion
    sig { returns(T::Array[T::Hash[String, String]) }
```

I've written this _way_ too many times. I need to write a type alias for it.

Reply:
Best I can do is `sig { returns(T::Array[T::Hash[String, T.any(Boolean, String)]) }` because replaces-base exists. Not sure if there are any other cases.